src/seedu/addressbook/data/person/Block.java

```diff
@@ -0,0 +1,12 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represents a block in a Person's address in the address book.
+ * Guarantees: immutable
+ */
+public class Block {
+    public final String value;
+    public Block(String block){
```

Comment:
You can consider moving the validation and tests of equality to the new classes you have created.